Automattic/woocommerce-subscriptions-core
# Deprecate upgrade scripts for versions before version 3

Fixes #150

## Description

⚠️ This code intentionally prevents database updates from happening when upgrading from older versions.

This PR deprecates upgrading procedures from versions prior to woocommerce-subscriptions-database version 3.0 (which was released Jan 2020.) Some code has been removed to prevent upgrades from running, where possible code has just been marked as deprecated. 

An admin notice has been added to warn users about the database upgrade.

![image](https://user-images.githubusercontent.com/57298/168213241-9cf420a4-3144-46e6-adca-d2808f4030bc.png)

Due to the removal of some of the really outdated upgrades I was able to disable the code that caused issue #155 

 **NB** Only the database updates are prevented from running. Other subscriptions functionality can still be used including adding & purchasing products.

## How to test this PR

1. On a clean WP/WC instance, install an old version of WooCommerce Subscriptions (prior to 3.0.0)
2. Deactivate WooCommerce Subscriptions
3. Build woocommerce-subscriptions-core, install & activate.
4. Install WooCommerce Payments. (to see Upgrade Notice)
5. Disable subscriptions-core. (this will use WCPay bundled version of subscriptions-core which will upgrade successfully)
6. Re-activate subscriptions-core

You may also want to check `woocommerce_subscriptions_active_version` during this testing to check that the version updates correctly.

## Product impact


This change will only affect those that have WooCommerce Subscriptions 2.6.5 or before installed. Prior to upgrading to the latest version of either WooCommerce Subscriptions or WooCommerce Payments.
